src/psyclone/dynamo0p3.py

```diff
@@ -9267,11 +9226,10 @@ def create(call, parent=None):
 
         # The kernel itself
         kern = DynKern()
-        kern.load(call)
+        kern.load(call, cloop.children[3])
```

Comment:
Is this a bug fix?

Reply:
Without giving the parent information at the begging the `self.root()` method just returns itself.

Note that there is probably a design issue here as well because `DynKern.__init__()` does not accept `parent=` nor doesn't call `super().__init__()` and therefore is not an really extension of the abstract `__init__()` of `Node` (?). Also the two step initialization with `load()` makes the object momentanly unusable between the `init` and the `load` (`self.parent` at that point causes an exception).